controllers/gatekeeper_controller.go

```diff
@@ -293,39 +296,21 @@ func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager, fromCPSMgrSour
 		Complete(r)
 }
 
-func (r *GatekeeperReconciler) initConfig(ctx context.Context, gatekeeper *operatorv1alpha1.Gatekeeper) error {
-	config := &v1alpha1.Config{}
+func (r *GatekeeperReconciler) StopSubControllers() {
+	log := r.Log.WithName("cleanup")
 
-	err := r.Get(ctx, types.NamespacedName{
-		Namespace: r.Namespace,
-		Name:      "config",
-	}, config)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			err = createDefaultConfig(ctx, r.Client, r.Namespace, gatekeeper, r.Scheme)
-			if err != nil {
-				return err
-			}
-
-			r.Log.Info("The Gatekeeper Config resource is created.")
-
-			return nil
-		}
-
-		return err
+	if r.cpsCtrlCtxCancel != nil {
+		log.Info("Stopping the ConstraintPodStatus controller manager.")
+		r.cpsCtrlCtxCancel()
 	}
 
-	// Add the default exempt namespace list
-	// disableDefaultMatches = true, only set the Gatekeeper CR config.matches values and
-	// the default exempt namespaces will not be appended
-	err = setExemptNamespaces(ctx, r.Client, config, gatekeeper, r.Scheme, r.Log)
-	if err != nil {
-		r.Log.V(1).Error(err, "Adding default exempt namespaces has failed")
-
-		return err
+	if r.configCtrlCtxCancel != nil {
+		log.Info("Stopping the Config controller manager.")
+		r.configCtrlCtxCancel()
 	}
 
-	return nil
+	r.subControllerWait.Wait()
+	log.Info("The subcontrollers are shutting down.")
```

Comment:
```suggestion
	log.Info("The extra controllers have shut down.")
```

Reply:
"extra" doesn't sound like a good thing. Maybe "resource controllers"?
I'd say "Gatekeeper resource controllers", but unfortunately the operator CRD is called Gatekeeper, so it's overloaded. 😞

Reply:
Sure, resource controllers works.